From 5c52de16b20af59170c634841392d654fe37dc7d Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Thu, 23 Apr 2020 15:06:34 +0200
Subject: [PATCH] debianize: Add placeholder for original version to
 CHANGELOG_V

This allows to append to the latest upstream version when using an
unpinned source version. The format of the placeholder is
"<orig-version>".

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 meta/classes/debianize.bbclass | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/meta/classes/debianize.bbclass b/meta/classes/debianize.bbclass
index da43a4e2..e77be232 100644
--- a/meta/classes/debianize.bbclass
+++ b/meta/classes/debianize.bbclass
@@ -11,6 +11,15 @@ DESCRIPTION ?= "must not be empty"
 MAINTAINER ?= "Unknown maintainer <unknown@example.com>"
 
 deb_add_changelog() {
+	changelog_v="${CHANGELOG_V}"
+	if [ -f ${S}/debian/changelog ]; then
+		if [ ! -f ${WORKDIR}/changelog.orig ]; then
+			cp ${S}/debian/changelog ${WORKDIR}/changelog.orig
+		fi
+		orig_version=$(dpkg-parsechangelog -l ${WORKDIR}/changelog.orig -S Version)
+		changelog_v=$(echo "${changelog_v}" | sed 's/<orig-version>/'${orig_version}'/')
+	fi
+
 	timestamp=$(find ${S}/ -type f -not -path "${S}/debian/*" -printf "%T@\n"|sort -n -r|head -n 1)
 	if [ -n "${timestamp}" ]; then
 		date=$(LANG=C date -R -d @${timestamp})
@@ -18,7 +27,7 @@ deb_add_changelog() {
 		date=$(LANG=C date -R)
 	fi
 	cat <<EOF > ${S}/debian/changelog
-${PN} (${CHANGELOG_V}) UNRELEASED; urgency=low
+${PN} (${changelog_v}) UNRELEASED; urgency=low
 
   * generated by Isar
 
@@ -26,7 +35,7 @@ ${PN} (${CHANGELOG_V}) UNRELEASED; urgency=low
 EOF
 	if [ -f ${WORKDIR}/changelog ]; then
 		if head -1 "${WORKDIR}"/changelog | \
-			grep -q -e "^${PN} (${CHANGELOG_V})"
+			grep -q -e "^${PN} (${changelog_v})"
 		then
 			# entry for our version already there, use unmodified
 			rm ${S}/debian/changelog
-- 
2.16.4

